src/v/hashing/murmur.cc

```diff
@@ -109,6 +116,97 @@ uint32_t murmurhash3_x86_32(const void* key, std::size_t len, uint32_t seed) {
     return h1;
 }
 
+uint32_t murmurhash3_x86_32(const iobuf& data, uint32_t seed) {
+    uint32_t h1 = seed;
+
+    const uint32_t c1 = 0xcc9e2d51;
+    const uint32_t c2 = 0x1b873593;
+
+    //----------
+    // body
+
+    uint32_t torn_block;
+    auto torn_block_begin = reinterpret_cast<char*>(&torn_block);
+    auto torn_block_end = torn_block_begin + 4;
+    auto torn_block_data_end = torn_block_begin;
+
+    for (const auto& fragment : data) {
+        auto frag_begin = fragment.get();
```

Comment:
it seems like the body of this for-loop should be some kind of generic non-iobuf-specific interface that the iobuf adapter reuses.